boa_tester/src/exec/mod.rs

```diff
@@ -198,8 +198,12 @@ impl Test {
 
                 context.run_jobs();
 
-                if let Err(e) = async_result.inner.borrow().as_ref() {
-                    return (false, format!("Uncaught {e}"));
+                match *async_result.inner.borrow() {
+                    UninitResult::Err(ref e) => return (false, format!("Uncaught {e}")),
+                    UninitResult::Uninit if self.flags.contains(TestFlags::ASYNC) => {
+                        return (false, "async test didn't call `print`".to_string())
```

Comment:
Should we make the failed message a little more explicit? Something like "async test did not print \"Test262:AsyncTestComplete\""

Reply:
I like that!